routes/song.js

```diff
@@ -21,13 +21,17 @@ router.route('/:song_id')
     .get(function(req, res, next) {
         if (song.copyright === 'private') {
             if (!req.isAuthenticated()) {
+                //if the song is private and the user is not logged in, show no access
                 res.render('noaccess')
             } else if (song.contributor !== req.user.username && !passportFunction.isAdmin) {
+                //if the song is private and it is no the user's contributed song and the user is not admin, show no access
```

Comment:
might be better to say "if the user is the song contributor or admin, then show the song"
And then change the clause to match.
Otherwise by default do not show it.